### ISSUE TITLE
using Dropbox 8.1 MetaData is not subscriptable 

### ISSUE THREAD

**Issue by riccardodivirgilio** [jschneier/django-storages#396]
https://github.com/jschneier/django-storages/blob/adefe324be22843148b03a2068d2922eaae1a48b/storages/backends/dropbox.py#L82

using python3.5 and dropbox 8.1 I have got the following error:

```
  File "/home/server/env/lib/python3.5/site-packages/storages/backends/dropbox.py", line 79, in listdir
    for entry in metadata['contents']:
TypeError: 'FolderMetadata' object is not subscriptable
```

if I open the source code of dropbox I can see that the class MetaData is not defining a __getitem__ method which is why this code is failing...

thanks and good job with this package

**Comment by riccardodivirgilio** [comment]
what version of dropbox is supported from this code? thanks

**Comment by danni** [comment]
I tried this with Dropbox 8.0.0, which is what's in the tests. I suspect this may actually be related to Python 3.

**Comment by patroqueeet** [comment]
I ran in the same issue with python 2.7:

```
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/dbbackup/utils.py", line 111, in wrapper
    func(*args, **kwargs)
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/dbbackup/management/commands/dbbackup.py", line 63, in handle
    self._cleanup_old_backups(database=database_key)
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/dbbackup/management/commands/_base.py", line 132, in _cleanup_old_backups
    servername=servername)
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/dbbackup/storage.py", line 237, in clean_old_backups
    servername=servername)
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/dbbackup/storage.py", line 123, in list_backups
    files = [f for f in self.list_directory() if utils.filename_to_datestring(f)]
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/dbbackup/storage.py", line 78, in list_directory
    return self.storage.listdir(path)[1]
  File "/home/darg/darg/.ve/local/lib/python2.7/site-packages/storages/backends/dropbox.py", line 141, in listdir
    for entry in metadata['contents']:
```

**Comment by danni** [comment]
Yep, what I learned is that the API had completed changed, and this was being masked in the tests by the mocks.

**Comment by patroqueeet** [comment]
@danni how did you fix that?

**Comment by patroqueeet** [comment]
fixed this in `backends/dropbox.py` with replacing `listdir()` with 

```
    def listdir(self, path):
        directories, files = [], []
        full_path = self._full_path(path)
        metadata = self.client.files_get_metadata(full_path)
        import pdb; pdb.set_trace()
        # for entry in metadata['contents']:
        for entry in self.client.files_list_folder(full_path).entries:
            entry.path_lower = entry.path_lower.replace(full_path, '', 1)
            entry.path_lower = entry.path_lower.replace('/', '', 1)
            if isinstance(entry, FolderMetadata):
                directories.append(entry.path_lower)
            else:
                files.append(entry.path_lower)
        return directories, files
```

sorry for not sending a PR, but I'm under heavy fire here and don't want to promise something I can't deliver.

**Comment by patroqueeet** [comment]
ah! stupid me. just now saw the PR from @danni . SRY! good job!

**Comment by jschneier** [comment]
Fixed by #724.
